### DISCUSSION TITLE
Button to open a URL in a property

### DISCUSSION THREAD

**Discussion by Eflores89** [chrisgurney/obsidian-note-toolbar#52] [Q&A] [answered]
I have pages with a URL property. Such as: 

```
---
url: www.google.com
---
```

I hide all properties by default on all my notes, but i'm trying to build a toolbar for notes in the folder that contains links so that when I click a button, it will take me to the URL (google.com in this example, via my browser).

Any suggestions on how to build this? I don't see any programmatic way to get page properties.

**Comment by chrisgurney** [comment]
@Eflores89 For simple URLs (without slashes, and potentially other non-alphanumerics) this should work:

- create a new item and give it an icon and/or label
- set it to a `URI` type
- enter this: `https://{{prop_url}}`

There's a little [more to read here](https://github.com/chrisgurney/obsidian-note-toolbar/wiki/Variables).

`{{prop_url}}` won't work by itself in your case as it could technically be a note name.

If you have the protocol in your `url` property, such as `https://www.google.com`, I think the plugin is attempting to encode it, even though perhaps it should keep it in that case. I'll look into that.

**Reply by chrisgurney** [reply, marked as answer]
@Eflores89 Update: I'm going to remove encoding in the next version, and completely trust that the user knows what they're doing with their properties and links. 😄 

The way it **will** work with the next release will be as follows: 

If your property is called `url` and `url` = `google.com` then: 

- `{{prop_url}}` alone will create a note called `google.com`
- To make it a link, either:
  - add the protocol in the toolbar item:`https://{prop_url}}` or 
  - put the `https://` in the property itself: `url` = `https://google.com`

**Comment by Eflores89** [comment]
Oh I had not seen there were variables! Very cool! 

Also, sorry I wasn't clear, I do have the url it in the form `https://google.com`, but I agree with your pov on letting users figure out the encoding. I'll wait for next update then! Thanks!

**Comment by chrisgurney** [comment]
Version `1.6.0-beta` is now available (via BRAT) that includes the change to variable encoding (i.e., not encoding any more). Any help with testing is appreciated!
